### PR TITLE
FURNACE-57: Provide a way to extend the FurnaceDeploymentScenarioGene…

### PR DESCRIPTION
**Pull request by mbriskar**
…rator to alter the deployment of the furnace test

### CODE REVIEW

File: test-harness/arquillian/core/src/main/java/org/jboss/forge/arquillian/services/FurnaceAddonDeploymentEnhancer.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * An interface used by the {@link java.util.ServiceLoader} to alter the deployment of the test addons.
+ * This method will be called for every single furnace test that has this enhancer implementation on classpath.
+ * For example, it is possible to deploy default addon for each furnace test.
+ *
+ * @author <a href="mailto:mbriskar@gmail.com">Matej Briškár</a>
+ */
+public interface FurnaceAddonDeploymentEnhancer
+{
+    List<DeploymentDescription> enhanceDeployment(List<DeploymentDescription> deployments);
+}
```
Comment (gastaldi) [inline test-harness/arquillian/core/src/main/java/org/jboss/forge/arquillian/services/FurnaceAddonDeploymentEnhancer.java]: What happens if multiple enhancers are defined? Could you add some tests?

Comment (mbriskar) [inline test-harness/arquillian/core/src/main/java/org/jboss/forge/arquillian/services/FurnaceAddonDeploymentEnhancer.java]: they are all called, however their order is random

File: test-harness/arquillian/core/src/main/java/org/jboss/forge/arquillian/services/FurnaceAddonDeploymentEnhancer.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package org.jboss.forge.arquillian.services;
+
+import org.jboss.arquillian.container.spi.client.deployment.DeploymentDescription;
+
+import java.util.List;
+
+/**
+ * An interface used by the {@link java.util.ServiceLoader} to alter the deployment of the test addons.
```
Comment (gastaldi) [inline test-harness/arquillian/core/src/main/java/org/jboss/forge/arquillian/services/FurnaceAddonDeploymentEnhancer.java]: used by FurnaceDeploymentScenarioGenerator

**Comment by gastaldi** [comment]
Merged, thanks

**Pull request closed without merging**